beetsplug/missing.py

```diff
@@ -106,35 +111,95 @@ def __init__(self):
         self._command.parser.add_option(
             u'-t', u'--total', dest='total', action='store_true',
             help=u'count total of missing tracks')
+        self._command.parser.add_option(
+            u'-a', u'--albums', dest='albums', action='store_true',
+            help=u'show missing albums for artist instead of tracks')
         self._command.parser.add_format_option()
 
     def commands(self):
         def _miss(lib, opts, args):
             self.config.set_args(opts)
-            count = self.config['count'].get()
-            total = self.config['total'].get()
-            fmt = config['format_album' if count else 'format_item'].get()
+            albms = self.config['albums'].get()
 
-            albums = lib.albums(decargs(args))
-            if total:
-                print(sum([_missing_count(a) for a in albums]))
-                return
+            helper = self._missing_albums if albms else self._missing_tracks
+            helper(lib, args)
 
-            # Default format string for count mode.
-            if count:
-                fmt += ': $missing'
+        self._command.func = _miss
+        return [self._command]
 
-            for album in albums:
-                if count:
-                    if _missing_count(album):
-                        print_(format(album, fmt))
+    def _missing_tracks(self, lib, args):
```

Comment:
Instead of taking a parameter called `args`, which is bytestrings, these two functions should probably take a query called `query`—the result of `decargs(args)`. This should make the method signatures a little clearer.